From b306532e7a4a3f235b24e5dab22ab36b849ac886 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Wed, 16 Apr 2025 11:50:37 +0100
Subject: [PATCH] generator: Use new FDevice type for the pvs-full pv_name
 field

Remove the code which did explicit reverse device name translation,
and use the generator's code instead.

(cherry picked from commit 5a16d1120fb7b046974abde43b1c40250bfd1a95)
---
 daemon/lvm.c         | 29 +----------------------------
 generator/structs.ml |  2 +-
 2 files changed, 2 insertions(+), 29 deletions(-)

diff --git a/daemon/lvm.c b/daemon/lvm.c
index b8c01f71..7e76e17c 100644
--- a/daemon/lvm.c
+++ b/daemon/lvm.c
@@ -146,34 +146,7 @@ do_vgs (void)
 guestfs_int_lvm_pv_list *
 do_pvs_full (void)
 {
-  guestfs_int_lvm_pv_list *r;
-  size_t i;
-  char *din, *dout;
-
-  r = parse_command_line_pvs ();
-  if (r == NULL)
-    /* parse_command_line_pvs has already called reply_with_error */
-    return NULL;
-
-  /* The pv_name fields contain device names which must be reverse
-   * translated.  The problem here is that the generator does not have
-   * a "FMountable" field type in types.mli.
-   */
-  for (i = 0; i < r->guestfs_int_lvm_pv_list_len; ++i) {
-    din = r->guestfs_int_lvm_pv_list_val[i].pv_name;
-    if (din) {
-      dout = reverse_device_name_translation (din);
-      if (!dout) {
-        /* reverse_device_name_translation has already called reply_with_error*/
-        /* XXX memory leak here */
-        return NULL;
-      }
-      r->guestfs_int_lvm_pv_list_val[i].pv_name = dout;
-      free (din);
-    }
-  }
-
-  return r;
+  return parse_command_line_pvs ();
 }
 
 guestfs_int_lvm_vg_list *
diff --git a/generator/structs.ml b/generator/structs.ml
index c1408804..93a3ed03 100644
--- a/generator/structs.ml
+++ b/generator/structs.ml
@@ -35,7 +35,7 @@ type struc = {
  * we have to pull out the LVM columns separately here.
  *)
 let lvm_pv_cols = [
-  "pv_name", FString;
+  "pv_name", FDevice;
   "pv_uuid", FUUID;
   "pv_fmt", FString;
   "pv_size", FBytes;
